app/code/Magento/Inventory/etc/di.xml

```diff
@@ -94,4 +94,10 @@
     <!-- Query interfaces -->
     <preference for="Magento\InventoryApi\Api\GetProductQuantityInStockInterface" type="Magento\Inventory\Model\GetProductQuantityInStock"/>
     <preference for="Magento\InventoryApi\Api\IsProductInStockInterface" type="Magento\Inventory\Model\IsProductInStock"/>
+    <preference for="Magento\Inventory\Model\StockIndexTableNameResolverInterface" type="Magento\Inventory\Model\StockIndexTableNameResolver"/>
```

Comment:
Also need to move to `InventoryIndexer`

Reply:
Can not move it, because it is used in InventoryCatalog module and InventoryIndexer has dependepcy on InventoryCatalog.

Reply:
Need to delete dependency `InventoryIndex` on `InventoryCatalog`
Just move related files from setup to `InventoryCatalog` setup

Reply:
Done.